packages/editor-ui/src/components/ExecutionsUsage.vue

```diff
@@ -0,0 +1,183 @@
+<template>
+	<div :class="$style.container">
+		<div v-if="!isTrialExpired && trialHasExecutionsLeft" :class="$style.usageText">
+			<i18n path="executionUsage.currentUsage">
+				<template #text>
+					<n8n-text size="xsmall" color="text-dark">
+						{{ locale.baseText('executionUsage.currentUsage.text') }}
+					</n8n-text>
+				</template>
+				<template #count>
+					<n8n-text size="xsmall" :bold="true" color="warning">
+						{{
+							locale.baseText('executionUsage.currentUsage.count', {
+								interpolate: { days: daysLeftOnTrial.toString() },
+							})
+						}}
+					</n8n-text>
+				</template>
+			</i18n>
+		</div>
+		<div v-if="isTrialExpired" :class="$style.usageText">
+			<n8n-text size="xsmall" color="danger">
+				{{ locale.baseText('executionUsage.expired.text') }}
+			</n8n-text>
+		</div>
+		<div v-if="!trialHasExecutionsLeft" :class="$style.usageText">
+			<n8n-text size="xsmall">
+				{{ locale.baseText('executionUsage.ranOutOfExecutions.text') }}
+			</n8n-text>
+		</div>
+		<div v-if="!isTrialExpired" :class="$style.usageCounter">
+			<div>
+				<progress
+					:class="[
+						trialHasExecutionsLeft ? $style.progressBarSuccess : $style.progressBarDanger,
+						$style.progressBar,
+					]"
+					:value="currentExecutions"
+					:max="maxExecutions"
+				></progress>
+			</div>
+			<div :class="$style.executionsCountSection">
+				<n8n-text size="xsmall" :color="trialHasExecutionsLeft ? 'text-dark' : 'danger'">
+					{{ currentExecutions }}/{{ maxExecutions }}
+				</n8n-text>
+				<n8n-text size="xsmall" :color="trialHasExecutionsLeft ? 'text-dark' : 'danger'">{{
+					locale.baseText('executionUsage.label.executions')
+				}}</n8n-text>
+			</div>
+		</div>
+
+		<div :class="$style.upgradeButtonSection">
+			<n8n-button
+				:label="locale.baseText('executionUsage.button.upgrade')"
+				size="mini"
+				icon="gem"
+				type="success"
+				:block="true"
+				@click="onUpgradeClicked"
+			/>
+		</div>
+	</div>
+</template>
+
+<script setup lang="ts">
+import { onMounted, computed } from 'vue';
+import { i18n as locale } from '@/plugins/i18n';
+import { DateTime } from 'luxon';
+
+export interface CloudPlanData {
+	planSpec: PlanSpec;
+	instance: Instance;
+	usage: Usage;
+}
+
+export interface PlanSpec {
+	planId: number;
+	monthlyExecutionsLimit: number;
+	activeWorkflowsLimit: number;
+	credentialsLimit: number;
+	isActive: boolean;
+	displayName: string;
+	expirationDate: string;
+	metadata: PlanMetadata;
+}
+export interface PlanMetadata {
+	version: 'v1';
+	group: 'opt-out' | 'opt-in';
+	slug: 'pro-1' | 'pro-2' | 'starter' | 'trial-1';
+	trial?: Trial;
+}
+export interface Trial {
+	length: number;
+	gracePeriod: number;
+}
+export interface Instance {
+	createdAt: string;
+}
+export interface Usage {
+	executions: number;
+	activeWorkflows: number;
+}
```

Comment:
why are you duplicating the types here?

Reply:
damn did not know this.. maybe we can just extend the original types then so we are not duplicating the types completely.. @OlegIvaniv can you take a look here?
```
interface CloudPlanLocal extends CloudPlan {}
```